src/zkchannels/merchant/manage.rs

```diff
@@ -43,33 +88,23 @@ impl Command for List {
                     Cell::new(channel.status),
                 ]);
             }
-
-            println!("{}", table);
+            Ok(table.to_string())
         }
-        Ok(())
     }
 }
 
 #[async_trait]
 impl Command for Show {
-    async fn run(self, config: Config) -> Result<(), anyhow::Error> {
+    type Output = String;
+
+    async fn run(self, config: Config) -> Result<Self::Output, anyhow::Error> {
         let database = database(&config)
             .await
             .context("Failed to connect to local database")?;
         let details = database.get_channel_details_by_prefix(&self.prefix).await?;
 
         if self.json {
-            println!(
-                "{}",
-                json!({
-                    "channel_id": format!("{}", details.channel_id),
-                    "status": format!("{}", details.status),
-                    "contract_id": format!("{}", details.contract_id),
-                    "merchant_deposit": format!("{}", Amount::from(details.merchant_deposit)),
-                    "customer_deposit": format!("{}", Amount::from(details.customer_deposit)),
-                })
-                .to_string()
-            );
+            Ok(serde_json::to_string(&PublicChannelDetails::from(details))?)
```

Comment:
I love it, this is so much better than the printing!

Reply:
Yay! I just propagated this change to both `List` commands as well, for consistency.